### DISCUSSION TITLE
Change WiFi connection using web interface and access point

### DISCUSSION THREAD

**Discussion by TekyTech** [vicwomg/pikaraoke#331] [Ideas]
Connecting a karaoke installed on Raspberry Pi to a new WiFi is currently pretty cumbersome, especially a full desktop version is not installed. 

If I take it everywhere it is pretty awkward to carry a laptop with a card reader just to connect karaoke to another wireless. Or connect a keyboard while it’s supposed yo be designed to be controlled via web interface using a smart phone. 

So why all this hassle and not a simple way how it’s done in many other devices. 

When there’s no internet connection Raspberry can act as an access point on its own. Then any smart phone can connect to it even without internet access and it can access web interface of the karaoke. 

It would be great to be able to provide a new SSID (or search for available SSIDs around) and a password and it’s done! Might need only a restart. 

Then there would be no need to mess with external keyboard or worse take the SD card out to plug to a computer.

Everything can be done from the same web interface to reconnect to any new WiFi at any time without the need to mess with any extra hardware. 

Then this great toy can be carried to new places and used anywhere much easier. Then it becomes truly lightweight, compact and mobile. Easier to use too.

**Comment by vicwomg** [comment]
The readme covers this very topic. There's a package called RaspiWifiConfig that can do this. 

https://github.com/vicwomg/pikaraoke?tab=readme-ov-file#i-brought-my-pikaraoke-to-a-friends-house-and-it-cant-connect-to-their-network-how-do-i-change-wifi-connection-without-ssh

**Reply by BentWrench** [reply]
Hey, you can console in to the Pi and use the Network Manager TUI. This doesn't work through the web interface, but it is easier than editing the config files on the SD card.

To launch, enter:

`sudo nmtui`

Select Edit a connection
Tab/Arrow key down to the Wi-Fi section, then tab key over to <Add>
Configure your new network. It will require this info:

- Profile name
- SSID
- Security type
- Password

Enter all that then tab down to <OK>
Your new profile should show up on the list of Wi-Fi networks, if so, tab over to <Back>, press enter
Arrow key to Quit and then tab to <OK> to exit

You can do all this with a USB keyboard, or pre-configure it before you go somewhere.

**Comment by TekyTech** [comment]
I’m sorry there might have been misunderstanding of the point that I’m making. My whole post is exactly in response to that section titled “I brought my pikaraoke to a friend's house and it can't connect to their network. How do I change wifi connection without ssh?”  
It seems you might have missed my whole point why I wrote it and what I suggested. 

I could even elaborate. 
just like there’s a QR code to connect to the karaoke interface when it’s connected to the Internet, there could be two QR codes when it’s not connected. 

One QR code to connect to the local Pi’s access point WiFi. And the second to connect to the web interface local ip address.

then have it straight in they interface an ability to change raspberry WiFi settings to make it connect to another WiFi. 

Not connecting to raspberry’s desktop interface, not using the keyboard, not using ssh, not messing with taking SD out to plug into SD card reader in a computer and manually editing done files. 

It would be so much better experience for most non technically sophisticated users who’d love to use it but not going to bother with those simple yet extra steps that make it complicated with many inexperienced users for whom we, the tech people, build those things to enjoy. 

Hope it helps. 

Thanks.

**Reply by vicwomg** [reply]
I was referring to this part of the readme:

`Finally, this package can set up your pi as a self-configuring wireless access point, but hasn't been updated in a while https://github.com/jasbur/RaspiWiFi`

If you set that up, it creates a ssid for connecting your phone to the pi, then a wizard for connecting to local wifi access points. It does a much better job of this than we could implement in pikaraoke, so my suggestion was to check that out.
